internal/lib/sandbox/sandbox.go

```diff
@@ -431,6 +456,52 @@ func (s *Sandbox) Stopped() bool {
 	return s.stopped
 }
 
+// NetworkStopped returns whether the network has been stopped
+func (s *Sandbox) NetworkStopped() bool {
+	return s.networkStopped
+}
+
+// SetNetworkStopped sets the sandbox network state as stopped
+// This should be set after a network stop operation succeeds,
+// so we don't double stop the network
+// if createFile is true, it creates a "network-stopped" file
+// in the infra container's persistent dir
+// this is used to track the network is stopped over reboots
+// returns an error if an error occurred when creating the network-stopped file
+func (s *Sandbox) SetNetworkStopped(createFile bool) error {
+	if s.networkStopped {
+		return nil
+	}
+	s.networkStopped = true
+	if createFile {
+		if err := s.createFileInInfraDir(sbNetworkStoppedFilename); err != nil {
+			return fmt.Errorf("failed to create state file in container directory. Restores may fail: %v", err)
+		}
+	}
+	return nil
+}
+
+func (s *Sandbox) createFileInInfraDir(filename string) error {
+	infra := s.InfraContainer()
+	_, err := os.Create(filepath.Join(infra.Dir(), filename))
+	return err
+}
+
+func (s *Sandbox) RestoreStopped() {
+	if s.fileExistsInInfraDir(sbStoppedFilename) {
+		s.stopped = true
+	}
+	if s.fileExistsInInfraDir(sbNetworkStoppedFilename) {
+		s.networkStopped = true
+	}
+}
+
+func (s *Sandbox) fileExistsInInfraDir(filename string) bool {
+	infra := s.InfraContainer()
+	_, err := os.Stat(filepath.Join(infra.Dir(), filename))
```

Comment:
we should log if it is some other error.

Reply:
fixed